### DISCUSSION TITLE
escape sequence in output

### DISCUSSION THREAD

**Discussion by SolitudeSF** [kovidgoyal/kitty#4948] [General]
Is "end of user input" sequence being included into show_last_command_output intentional?

**Comment by kovidgoyal** [comment]
What's "end of user input"? EOF? It's probably coming from the pager.

**Reply by SolitudeSF** [reply]
i mean, when i do `printf "\e]133;C;\e\\"; echo "12312312312312"; printf "\e]133;A;cl=m;aid=1\e\\"`, show_last_command_output shows contents of first prinf too, not only result of echo.

**Comment by kovidgoyal** [comment]
that will be because whatever program you are using as a pager is not
filtering out OSC codes, unlike less. kitty passes hyperlink and prompt
mark codes to the pager so it can potentially use them.

**Reply by SolitudeSF** [reply]
yes, i know, im just asking if the part that is not actually part of command output supposed to be in the command output.

**Comment by kovidgoyal** [comment]
Its the escape code that marks the start of the output region.

**Reply by SolitudeSF** [reply]
... i know. im asking why is that included in command output if its not part of the command output.

**Comment by kovidgoyal** [comment]
... because it marks the start of the output.

**Reply by SolitudeSF** [reply]
so, by your logic, sequence marking end of the output should be included aswell.

**Reply by SolitudeSF** [reply]
but even if you disregard that, what is the practicality of doing that?

**Comment by kovidgoyal** [comment]
On Sun, Apr 10, 2022 at 09:49:24AM -0700, Solitude wrote:
> so, by your logic, sequence marking end of the output should be included aswell.

There isnt one.

**Reply by SolitudeSF** [reply]
N/A/P commands act as one.

**Comment by kovidgoyal** [comment]
On Sun, Apr 10, 2022 at 09:50:03AM -0700, Solitude wrote:
> but even if you disregard that, what is the practicality of doing that

Whats the downside?

**Reply by SolitudeSF** [reply]
practically, probably none, im just disagreeing with semantics. these bytes were not output by the commands, so they are by definition arent command output.

**Comment by SolitudeSF** [comment]
whatever, forget about it. it doesnt affect my workflow, im just annoyed, because i dont see the logic.

**Reply by kovidgoyal** [reply]
There's no real logic to it. When outputting the full scrollback these are obviously useful, so they are present in the buffer/common code. When showing a single commands output they are not useful but they are also not harmful, so I havent bothered to remove them.
